src/fairseq2/recipes/lm/text_generate.py

```diff
@@ -332,26 +349,32 @@ class TextGenerateUnit(AbstractGeneratorUnit[SequenceBatch]):
 
     _generator: SequenceGenerator
     _text_decoder: TextTokenDecoder
-    _output_stream: Optional[TextIO]
+    _output_txt_txt_stream: Optional[TextIO]
+    _output_jsonl_txt_stream: Optional[TextIO]
     _metric_bag: SequenceGenerationMetricBag
 
     def __init__(
         self,
         generator: SequenceGenerator,
         tokenizer: TextTokenizer,
         gang: Gang,
-        output_stream: Optional[TextIO],
+        output_txt_txt_stream: Optional[TextIO],
+        output_jsonl_txt_stream: Optional[TextIO],
     ) -> None:
         super().__init__(generator.model)
 
         self._generator = generator
 
         self._text_decoder = tokenizer.create_decoder()
 
-        self._output_stream = output_stream
+        self._output_txt_txt_stream = output_txt_txt_stream
+
+        self._output_jsonl_txt_stream = output_jsonl_txt_stream
 
         self._metric_bag = SequenceGenerationMetricBag(gang)
 
+        self._gang = gang
```

Comment:
this made debugging much easier as I can access gang from the `__call__`.

Reply:
Just curious, how were you using `gang` within `__call__` during debugging? Wondering whether we can streamline it.

Reply:
maybe there is a better way, what I did during the run

```
if self._gang.rank == 0: import pdb; pdb.set_trace()

self._gang.barrier()
```

is there a better way to do this without having the gang object in the scope?

Reply:
You might consider using the global `get_rank()` function defined [here](https://github.com/facebookresearch/fairseq2/blob/main/src/fairseq2/gang.py#L572). It is not dependent on `Gang` and can be used pretty much anywhere (e.g. deep inside a call stack)